upgrade/upgrade-scripts/upgrade

```diff
@@ -446,6 +448,10 @@ function finalize() {
 	rm -rf "$IMAGE_PATH" || die "failed to remove unpacked upgrade image"
 
 	remove_upgrade_properties
+
+	if [[ -f "$HOTFIX_PATH" ]]; then
+		rm -f "$HOTFIX_PATH" || die "failed to remove hotfix file"
```

Comment:
Can you comment on why you chose to put this logic here? e.g. why not within the `execute` script?

For not-in-place upgrades, where we create a new root filesystem, do we still want to remove this file from the old/current root filesystem? I ask, because for that type of upgrade, I don't think we would have modified the old/current root filesystem.. and we may still use it, in the case of a rollback.. at that point, would we want the file to still be there, or to have been removed?

Additionally, if we end up using these script for application of a hotfix, do we still want to remove this file after the "upgrade"? E.g. I'm imagining at some point, we may deliver hotfixes as an "upgrade image", and apply the hotfix similar to how we perform deferred upgrades.. in such an architecture, would this logic still apply?

Reply:
Ya I put it here specifically since it should be beyond the point of no return as far as rollbacks are concerned. I do not want to remove this file if there is still a possibility of a rollback happening. This of course means that I think the original hotfix file is still relevant post-rollback. Let me know if you think otherwise.

If and when we start using these scripts to perform hotfix install, I would imagine the same location to still be the right place to add in new logic to append the new hotfix data into the file - rather than deleting it as in the upgrade case. This is assuming the finalize() function is relevant for the hotfix install workflow. If not, we can find a different location for the hotfix specific logic.

Reply:
> Let me know if you think otherwise.

Ah yes, I see now.. OK, yea, I think this location should work then. Thanks for confirming.

> I would imagine the same location to still be the right place to add in new logic to append the new hotfix data into the file

Sounds good to me; thanks for thinking that case through.